### PR TITLE
chore(docs): migrate cloud docs to .com

### PR DESCRIPTION
**Pull request by MarcusCole518**
## Description

This PR is to merge the following docs from our Gatsby Cloud Knowledge Base to the gatsbyjs.com docs.

*Links are to current Cloud Docs*

1. Create a Site from a Template: https://support.gatsbyjs.com/hc/en-us/articles/360059155134-Create-a-Site-from-a-Template
2. Create a Site from a Repository: https://support.gatsbyjs.com/hc/en-us/articles/360059253654-Create-a-Site-from-a-Repository
3. Build and Preview Webhooks: https://support.gatsbyjs.com/hc/en-us/articles/360052324394-Build-and-Preview-Webhooks
4. Deploying to Gatsby Cloud Hosting: https://support.gatsbyjs.com/hc/en-us/articles/360061641114-Deploying-to-Gatsby-Cloud-Hosting
5. Setting a Node.js Version for your Gatsby Cloud Site: https://support.gatsbyjs.com/hc/en-us/articles/360052876333-How-do-I-set-a-Node-js-version-for-my-site-
6. Apex Domains and "www" Subdomains: https://support.gatsbyjs.com/hc/en-us/articles/360061651754-Apex-Domains-and-www-Subdomains
7. Remove Trailing Slash in Netlify URLs: https://support.gatsbyjs.com/hc/en-us/articles/360056349113-Remove-trailing-slash-in-Netlify-URLs
8. Outgoing Notifications: https://support.gatsbyjs.com/hc/en-us/articles/1500000480181-Outgoing-Notifications

### CODE REVIEW

File: docs/docs/reference/cloud/apex-and-subdomains.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+### Adding the gatsbyjs.io domain to a redirect
+
+Gatsby Cloud now allows you to redirect your `gatsbyjs.io` domain to your primary domain. This ensures that search engines don't index the `gatsbyjs.io` domain. To do this:
+
+1. Next to where your `gatsbyjs.io` domain is listed, Select **Primary Redirect**.
+2. Select a Primary domain:
+
+![Gatsby Domain Redirect](../../images/gatsby-domain-redirect.png)
+
+All `gatsbyjs.io` domains will 301 redirect to the primary domain that you select.
```
Comment (LekoArts) [inline docs/docs/reference/cloud/apex-and-subdomains.md:43]: Explain what a 301 redirect is

File: docs/docs/reference/cloud/build-and-preview-webhooks.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+---
+title: "Build and Preview Webhooks"
+description: "A webhook is an action that runs when the application providing the webhook is notified. The webhook has a URL which can receive HTTP requests (usually POST). A request sent to this URL lets the application know that an event has happened and it may or may not contain a message."
```
Comment (LekoArts) [inline docs/docs/reference/cloud/build-and-preview-webhooks.md:3]: That description is too long, needs to be more concise

File: docs/docs/reference/cloud/outgoing-notifications.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
+---
+title: "Outgoing Notifications"
```
Comment (LekoArts) [inline docs/docs/reference/cloud/outgoing-notifications.md:2]: This is not a reference guide but a how-to guide

**Review by LekoArts — CHANGES_REQUESTED** [review]
First pass on this, I'd need to dig into the details later

File: docs/docs/how-to/cloud/create-site-from-repository.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+![Import from a Repository](../../images/import-from-repo.png)
+
+### 3. Specify repository details
+
+Once connected to your Git provider, find the repository you would like to add to Gatsby Cloud and click **Import**. You can then provide the details for your site:
+
+1. The site name (defaults to `<repo name>-<branch name>`).
+2. The branch to import from and set as your "production" branch in Gatsby Cloud (in this case, the `main` branch).
+3. The base directory, i.e., the directory containing the Gatsby site (the root directory, `/`, is default).
```
[the remark this answers is not in the document]
Comment (MarcusCole518) [reply docs/docs/how-to/cloud/create-site-from-repository.md]: thanks for the tip! This worked well on the subsections but didn't seem to work for the h3 steps; each step was still listed as "1"

File: docs/docs/how-to/cloud/create-site-from-repository.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+## Directions
+
+### 1. Select a git provider
+
+From your Gatsby Cloud dashboard, click the **Add a Site** button.
+
+![Button to add a site from the Gatsby Cloud dashboard](../../images//add-site.png)
+
+In the "Import from a Git repository" section, choose your Git provider and install the Gatsby Cloud app. For the sake of this article you'll use GitHub, however GitLab and Bitbucket are also supported.
```
Comment (LekoArts) [inline docs/docs/how-to/cloud/create-site-from-repository.md:24]: The proper noun for GitHub is `GitHub` and in some other places of the added docs we have `Github`. So make sure that those are all consistent and correct 👍  Same applies to `GitLab` and `BitBucket`

File: docs/docs/how-to/cloud/create-site-from-repository.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+## Prerequisites
+
+You can use the [gatsby-starter-blog](https://github.com/gatsbyjs/gatsby-starter-blog) as a starting point. Select the "Use this template" button to create a new repository using the starter as a template.
+
+![Button to create a new repository from a starter](../../images/use-this-template.png)
+
+## Directions
+
+### 1. Select a git provider
```
Comment (LekoArts) [inline docs/docs/how-to/cloud/create-site-from-repository.md:18]:
```suggestion
### 1. Select a Git provider
```

Maybe also in other places, didn't check :)

File: docs/docs/how-to/cloud/create-site-from-repository.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+From your Gatsby Cloud dashboard, click the **Add a Site** button.
+
+![Button to add a site from the Gatsby Cloud dashboard](../../images//add-site.png)
```
Comment (LekoArts) [inline docs/docs/how-to/cloud/create-site-from-repository.md:22]:
```suggestion
![Button to add a site from the Gatsby Cloud dashboard](../../images/add-site.png)
```

File: docs/docs/how-to/cloud/outgoing-notifications.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+1. A generic webhook.
+1. A webhook that sends a Slack message.
+1. Pull Request comments in GitHub or GitLab.
+
+By default, the Pull Request comments are already set up for existing and new sites in Gatsby Cloud.
+
+### Adding webhook notifications
+
+1. First, navigate to Site Settings. In the sidebar menu under "General", select **Outgoing Notifications**. Then click **Add Notification** and choose the Webhooks option.
```
Comment (LekoArts) [inline docs/docs/how-to/cloud/outgoing-notifications.md:16]: (This also applies to the other docs)

Similarly to my comment in https://github.com/gatsbyjs/gatsby/pull/36218#discussion_r934198384 we should keep it consistent how we denote buttons/things to click in the UI.

I'd change the sentence to this:

```suggestion
1. First, navigate to "Site Settings". In the sidebar menu under "General", select **"Outgoing Notifications"**. Then click **"Add Notification"** and choose the "Webhooks" option.
```

So use bolding where necessary and use `""` to denote the buttons. This is how it was done in https://github.com/gatsbyjs/gatsby/blob/aeae8f366a48a1ec7832dc61b70b719ec7f92cef/docs/docs/how-to/cloud/adding-a-custom-domain.md after my comments

File: docs/docs/how-to/cloud/outgoing-notifications.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+- deploy succeeded
+- deploy failed
+
+3. Paste your webhook into the Outgoing webhook URL field, choose an event to listen to, then click **Save**. If you want to trigger notifications for multiple events, add a new notification for each.
+
+![Button to add a notification](../../images/add-notification.png)
+
+#### Webhook message payload
+
+You can expect the message payload to appear as follows:
```
Comment (LekoArts) [inline docs/docs/how-to/cloud/outgoing-notifications.md:37]:
```suggestion
You can expect the message payload to appear as following:
```

File: docs/docs/how-to/cloud/outgoing-notifications.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+  "resourceType": "SITE",
+  "event": "BUILD_SUCCEEDED"
+}
+```
+
+### Adding slack notifications
+
+To take advantage of Slack notifications, you first need to create a custom Slack application to handle the webhook message.
+
+1. First, visit https://api.slack.com/apps?new_app=1 to add a new Slack app. Add one for a workspace where you have [appropriate permissions](https://slack.com/help/articles/201314026-Permissions-by-role-in-Slack#apps-integrations).
```
Comment (LekoArts) [inline docs/docs/how-to/cloud/outgoing-notifications.md:58]:
```suggestion
1. First, [add a new Slack app](https://api.slack.com/apps?new_app=1). Add one for a workspace where you have [appropriate permissions](https://slack.com/help/articles/201314026-Permissions-by-role-in-Slack#apps-integrations).
```

File: docs/docs/how-to/cloud/outgoing-notifications.md [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+
+3. Now, select an appropriate channel for your notifications and click **Allow:**
+
+![Slack App Permissions](../../images/slack-app-permissoin.png)
+
+After that, you will see a success message and a new webhook URL will be generated. Copy the webhook URL.
+
+![Example of a Webhook URL](../../images/webhook-url.png)
+
+4. Finally, return to Gatsby Cloud and navigate to the Outgoing Ntifications settings. Click **Add Notification** and choose the Slack option. Paste the Slack application's webhook URL that you previously copied into the Outgoing webhook URL field.
```
Comment (LekoArts) [inline docs/docs/how-to/cloud/outgoing-notifications.md:74]:
```suggestion
4. Finally, return to Gatsby Cloud and navigate to the "Outgoing Notifications" settings. Click **"Add Notification"** and choose the "Slack" option. Paste the Slack application's webhook URL that you previously copied into the Outgoing Webhook URL field.
```

File: docs/docs/how-to/cloud/set-node-version.md [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+## Introduction
+
+You can specify a Node.js version for site in Gatsby Cloud in two different ways, by environment variable or `.nvmrc` file. If you don't set a Node version, Gatsby cloud will use the following defaults:
+
+- The default Node.js version is 14.
+- If you are using npm 7, the default is 16.
+
+### Environment variable
+
+You can set your Node.js version using the `NODE_VERSION` environment variable inside Gatsby Cloud. Read more about setting environment variables [here](../../reference/cloud/managing-environment-variables).
```
Comment (LekoArts) [inline docs/docs/how-to/cloud/set-node-version.md:15]: "Here" is a bad link text: https://www.gatsbyjs.com/contributing/gatsby-style-guide/#use-clear-hyperlinks

```suggestion
You can set your Node.js version using the `NODE_VERSION` environment variable inside Gatsby Cloud. Read more about [setting environment variables](/docs/reference/cloud/managing-environment-variables).
```

File: docs/docs/reference/cloud/apex-and-subdomains.md [hunk 13]
```diff
@@ -0,0 +1,43 @@
+---
+title: "Apex and Subdomains"
+description: "How to configure your apex and subdomains in Gatsby Cloud"
+---
+
+## Introduction
+
+Adding a custom domain to your Gatsby Cloud site will add both the apex domain and "www" subdomain. This guide will show you how to configure these domains, and redirect between the two.
```
Comment (LekoArts) [inline docs/docs/reference/cloud/apex-and-subdomains.md:8]: Add links to explain what an "Apex" domain is

File: docs/docs/reference/cloud/apex-and-subdomains.md [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+
+### Adding the domain
+
+1. Add either an apex domain or `www` subdomain to your Gatsby Hosting configuration. You can read more about adding a domain in [Adding a Custom Domain](/docs/how-to/cloud/adding-a-custom-domain/).
+
+2. You should see a redirect pair of domains with the domain you entered as a primary domain.
+
+![List of connected domains](../../images/primary-domain.png)
+
+3. Follow the directions for adding [DNS records](cloudflare.com/learning/dns/dns-records/) for both of the domains that are listed. The apex domain will need an A record pointed to an IP address and the subdomain will need a CNAME pointed at your Gatsby Hosting default URL, `YOUR_SITE_PREFIX`.
```
Comment (LekoArts) [inline docs/docs/reference/cloud/apex-and-subdomains.md:18]:
```suggestion
3. Follow the directions for adding [DNS records](https://cloudflare.com/learning/dns/dns-records/) for both of the domains that are listed. The apex domain will need an A record pointed to an IP address and the subdomain will need a CNAME pointed at your Gatsby Hosting default URL, `YOUR_SITE_PREFIX`.
```

**Review by LekoArts — CHANGES_REQUESTED** [review]
Overall it's nearly there. Once the things I mentioned are resolved it would be good to be merged (can also happen in my absence)

File: docs/docs/how-to/cloud/create-site-from-repository.md [hunk 15]
```diff
@@ -0,0 +1,73 @@
+---
+title: "Create a Site from a Repository"
+description: "How to create a Gatsby Cloud site from a repository"
```
Comment (j0sh77) [inline docs/docs/how-to/cloud/create-site-from-repository.md:3]: Nit-picking: the description format in these docs are not consistent. Can we try to sync them up with "Learn how to..."?

```suggestion
description: "Learn how to create a Gatsby Cloud site from a repository"
```

**Review by j0sh77 — APPROVED** [review]
Looks good minus the nit-pick. Feel free to ignore and ping me when you're ready to merge. 🚀 
Thanks for working through this!

File: docs/docs/reference/cloud/build-and-preview-webhooks.md [hunk 16]
```diff
@@ -0,0 +1,49 @@
+---
+title: "Build and Preview Webhooks"
+description: "This guide outlines the functionality of and differences between Gatsby's Build and Preview webhooks"
```
Comment (j0sh77) [inline docs/docs/reference/cloud/build-and-preview-webhooks.md:3]:
```suggestion
description: "Learn about Gatsby's Build and Preview webhooks and their differences"
```

File: docs/docs/how-to/cloud/create-site-from-template.md [hunk 17]
```diff
@@ -0,0 +1,66 @@
+---
+title: "Create a Site from a Template"
+description: "How to create a Gatsby Cloud site from a template"
```
Comment (j0sh77) [inline docs/docs/how-to/cloud/create-site-from-template.md:3]:
```suggestion
description: "Learn how to create a Gatsby Cloud site from a template"
```

File: docs/docs/how-to/cloud/deploying-to-gatsby-cloud-hosting.md [hunk 18]
```diff
@@ -0,0 +1,38 @@
+---
+title: "Deploying to Gatsby Cloud Hosting"
+description: "This tutorial will show you how to deploy your site to Gatsby Cloud hosting."
```
Comment (j0sh77) [inline docs/docs/how-to/cloud/deploying-to-gatsby-cloud-hosting.md:3]:
```suggestion
description: "Learn how to deploy your site to Gatsby Cloud hosting."
```

File: docs/docs/reference/cloud/apex-and-subdomains.md [hunk 19]
```diff
@@ -0,0 +1,43 @@
+---
+title: "Apex and Subdomains"
+description: "How to configure your apex and subdomains in Gatsby Cloud"
```
Comment (j0sh77) [inline docs/docs/reference/cloud/apex-and-subdomains.md:3]:
```suggestion
description: "Learn how to configure your apex and subdomains in Gatsby Cloud"
```

**Review by j0sh77 — APPROVED** [review]

**Comment by j0sh77** [comment]
THANK YOU :shipit: 🚀

**Pull request merged**